powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/MetricsUtils.java

```diff
@@ -130,6 +130,20 @@ public static void withMetricLogger(final Consumer<MetricsLogger> logger) {
         }
     }
 
+    /**
+     * Provide and immediately flush a {@link MetricsLogger}. It uses the default namespace
+     * specified either on {@link Metrics} annotation or via POWERTOOLS_METRICS_NAMESPACE env var.
+     * It by default captures function_request_id as property if used together with {@link Metrics} annotation. It will also
+     * capture xray_trace_id as property if tracing is enabled.
+     *
+     * @param logger the MetricsLogger
+     *
+     * @deprecated use {@link MetricsUtils#withMetricsLogger} instead
+     */
+    public static void withMetricLogger(final Consumer<MetricsLogger> logger) {
```

Comment:
Does it make sense to add also `@Deprecated` annotation?

Reply:
Sure!